source/language/types.rst

```diff
@@ -649,18 +684,37 @@ Casting from float
 discarding the fractional part for integer-type targets). As noted above, 
 if the value is too large to be represented in the 
 target type the result is implementation-specific. 
-Casting a ``float`` value to an ``angle[m]`` is accomplished by first 
-performing a modulo 2π operation on the float value. The resulting value 
-is then converted to the nearest ``angle[m]`` possible. In the event of a 
-tie between two possible nearest values the result is the one with an even 
-least significant bit (*i.e.* round to nearest, ties to even).
+
+Casting a ``float[n]`` value to an ``angle[m]`` involves finding the nearest
+representable value modulo :math:`\text{float}_n(2\pi)`, where ties between two
+possible representations are resolved by choosing to have zero in the
+least-significant bit (*i.e.* round to nearest, ties to even).  Casting the
+floating-point values ``inf``, ``-inf`` and all representations of ``NaN`` to
+``angle[m]`` is not defined.
+
+For example, given the double-precision floating-point value::
+
+   // The closest double-precision representation of 2*pi.
+   const float[64] two_pi = 6.283185307179586
+   // For double precision, we have
+   //   (two_pi * (127./512.)) / two_pi == (127./512.)
+   // exactly.
+   float[64] f = two_pi * (127. / 512.)
+
+the result of the cast ``angle[8](f)`` should have the bitwise representation
+``"01000000"`` (which represents the exact angle
+:math:`2\pi\cdot\frac{64}{256} = \frac\pi2`), despite ``"00111111"``
+(:math:`2\pi\cdot\frac{63}{256}`) being equally close, because of the
+round-to-nearest ties-to-even behaviour.
```

Comment:
**question**: How did you decide on this convention?  Does it follow the same convention as other casts e.g. `float` -> `bit[n]`?